OpenRA.Mods.Common/Traits/World/PathFinder.cs

```diff
@@ -54,22 +55,24 @@ public class PathFinder : IPathFinder
 		static readonly List<CPos> EmptyPath = new List<CPos>(0);
 		readonly World world;
 		DomainIndex domainIndex;
-		bool cached;
+		bool domainIndexCached;
+		readonly ICacheStorage<PathCacheKey, List<CPos>> cacheStorage;
```

Comment:
`ICacheStorage` is another redundant indirection, we should remove that and change this to just `PathCacheStorage cacheStorage`.

Reply:
Looks like this is the only use of `PathCacheStorage` and there is a 1:1 mapping of `PathFinder` to `PathCacheStorage`, so we can probably just merge that into `Pathfinder` directly.

Reply:
Agree.